hal/src/nRF52840/ble_hal.cpp

```diff
@@ -1673,6 +1709,11 @@ int BleObject::ConnectionsManager::init() {
         LOG(ERROR, "os_timer_create() failed.");
         goto error;
     }
+    if (generateEcdhKeyPair() != SYSTEM_ERROR_NONE) {
```

Comment:
We should reconsider generating the local keypair only once. The newest BLE spec has the following:

> To protect a device's private key, a device should implement a method to
> prevent an attacker from retrieving useful information about the device's private
> key. For this purpose, a device should change its private key after every pairing
> (successful or failed). Otherwise, it should change its private key whenever S +
> 3F > 8, where S is the number of successful pairings and F the number of
> failed attempts since the key was last changed.

(BLUETOOTH CORE SPECIFICATION Version 5.2 | Vol 3, Part H 2.3.6 Repeated attempts)

Reply:
Will do in this way: 

> a device should change its private key after every pairing
(successful or failed)